src/sentry/workflow_engine/migration_helpers/alert_rule.py

```diff
@@ -120,13 +138,75 @@ def migrate_metric_data_condition(
     alert_rule_trigger_data_condition = AlertRuleTriggerDataCondition.objects.create(
         alert_rule_trigger=alert_rule_trigger, data_condition=data_condition
     )
-    return data_condition, alert_rule_trigger_data_condition
+    return detector_trigger, data_condition, alert_rule_trigger_data_condition
 
 
-def migrate_resolve_threshold_data_condition(alert_rule: AlertRule) -> DataCondition:
+def get_resolve_threshold(
+    detector_data_condition_group: DataConditionGroup, threshold_type: AlertRuleThresholdType
+) -> float:
     """
-    Create data conditions for rules with a resolve threshold
+    Helper method to get the resolve threshold for a Detector if none is specified on
+    the legacy AlertRule.
     """
+    detector_triggers = DataCondition.objects.filter(condition_group=detector_data_condition_group)
+    if detector_triggers.count() > 1:
+        # there is a warning threshold for this detector, so we should resolve based on it
+        warning_data_condition = detector_triggers.filter(
+            condition_result=DetectorPriorityLevel.MEDIUM
+        ).first()
+        resolve_threshold = warning_data_condition.comparison
+    else:
+        # critical threshold value
+        resolve_threshold = detector_triggers.first().comparison
+
+    # XXX (mifu67): I saw this workaround for off by one error on the original auto-resolution.
+    # Do we still need this? Copying the comment over:
+    # -------------------------
+    # Since we only support gt/lt thresholds we have an off-by-one with auto
+    # resolve. If we have an alert threshold of > 0, and no resolve threshold, then
+    # we'd automatically set this to < 0, which can never happen. To work around
+    # this, we add a small amount to the number so that in this case we'd have
+    # the resolve threshold be < 0.000001. This means that when we hit 0 we'll still
+    # resolve as expected.
+    if threshold_type == AlertRuleThresholdType.ABOVE.value:
+        resolve_threshold += 0.000001
+    else:
+        resolve_threshold -= 0.000001
+
+    return resolve_threshold
+
+
+def migrate_resolve_threshold_data_conditions(alert_rule: AlertRule) -> DataCondition:
+    """
+    Create data conditions for the old world's "resolve" threshold. If a resolve threshold
+    has been explicitly set on the alert rule, then use this as our comparison value. Otherwise,
+    we need to figure out what the resolve threshold is based on the trigger threshold values.
+    """
+    alert_rule_detector = AlertRuleDetector.objects.get(alert_rule=alert_rule)
```

Comment:
may also want the select_related here too, so you can select the detector + the detectors conditions.